### PR TITLE
Two fixes for our device tree

### PR DESCRIPTION
**Pull request by nsherbina1999**

### CODE REVIEW

**Review by Exhigh — COMMENTED** [review]
Refer: https://github.com/aosp-mirror/platform_frameworks_base/commit/802d773c03c5d2c25902c43be82ebece496347b3

which sort of shows the error isn't related to it being binderized or passed through

**Review by Exhigh — COMMENTED** [review]
This overlay is required for the wifi to do background scans to connect to networks while the device is in range of the network, those errors are just log spams from the driver, anyway

**Review by Exhigh — COMMENTED** [review]
It technically isn't the egl crashing its just catching an exception, but the exception comes from the fact that egl tries destroying the surface that doesn't exist anymore, since it faded already, sure we could stop the fade animation but no means are we having any performance benefit from having our device close earlier, neither does it help in waking the device quicker.

Tldr. It doesn't really matter but if everyone's, ok with this, we can have this added

```
02-02 16:53:40.404  3684  3718 E libEGL  : eglDestroySurfaceImpl:896 error 300d (EGL_BAD_SURFACE)
02-02 16:53:40.405 14337 14337 I wm_on_paused_called: [175929323,com.launcher.CustomLauncher,performPause]
02-02 16:53:40.405  3684  3718 E ColorFade: eglDestroySurface failed: error 12301
02-02 16:53:40.405  3684  3718 E ColorFade: java.lang.Throwable
02-02 16:53:40.405  3684  3718 E ColorFade:     at com.android.server.display.ColorFade.logEglError(ColorFade.java:700)
02-02 16:53:40.405  3684  3718 E ColorFade:     at com.android.server.display.ColorFade.destroyEglSurface(ColorFade.java:645)
02-02 16:53:40.405  3684  3718 E ColorFade:     at com.android.server.display.ColorFade.dismissResources(ColorFade.java:341)
02-02 16:53:40.405  3684  3718 E ColorFade:     at com.android.server.display.DisplayPowerState.dismissColorFadeResources(DisplayPowerState.java:207)
02-02 16:53:40.405  3684  3718 E ColorFade:     at com.android.server.display.DisplayPowerController.animateScreenStateChange(DisplayPowerController.java:1515)
02-02 16:53:40.405  3684  3718 E ColorFade:     at com.android.server.display.DisplayPowerController.updatePowerState(DisplayPowerController.java:861)
02-02 16:53:40.405  3684  3718 E ColorFade:     at com.android.server.display.DisplayPowerController.access$600(DisplayPowerController.java:92)
02-02 16:53:40.405  3684  3718 E ColorFade:     at com.android.server.display.DisplayPowerController$DisplayControllerHandler.handleMessage(DisplayPowerController.java:1942)
02-02 16:53:40.405  3684  3718 E ColorFade:     at android.os.Handler.dispatchMessage(Handler.java:106)
02-02 16:53:40.405  3684  3718 E ColorFade:     at android.os.Looper.loop(Looper.java:223)
02-02 16:53:40.405  3684  3718 E ColorFade:     at android.os.HandlerThread.run(HandlerThread.java:67)
02-02 16:53:40.405  3684  3718 E ColorFade:     at com.android.server.ServiceThread.run(ServiceThread.java:44)
```

**Comment by nsherbina1999** [comment]
https://github.com/Exynos7580/android_device_samsung_a3xeltexx/commit/37cb1b1d4213240f7d10c2ba5edd34b68660090f - hwbinder
https://github.com/universal5433/android_device_samsung_trelte-common/commit/25751430efc1f9c75f5d0b4493e25941c17bdf80 - EGL crash
https://github.com/aperomsik/lineageos_android_device_asus_grouper/commit/fc7cab79fc1c668a4f89e33c5afcb60bf8f99450 - Wi-Fi scanning

This is why I've created those commits.

**Comment by Exhigh** [comment]
> [Exynos7580/android_device_samsung_a3xeltexx@37cb1b1](https://github.com/Exynos7580/android_device_samsung_a3xeltexx/commit/37cb1b1d4213240f7d10c2ba5edd34b68660090f) - hwbinder

This commit seems to tackle another issue, not the particular one about the energy controller anyway.

> [universal5433/android_device_samsung_trelte-common@2575143](https://github.com/universal5433/android_device_samsung_trelte-common/commit/25751430efc1f9c75f5d0b4493e25941c17bdf80) - EGL crash

We could merge this commit like I mentioned above, accompanied by some test builds to make sure it fixes the issue.

> [aperomsik/lineageos_android_device_asus_grouper@fc7cab7](https://github.com/aperomsik/lineageos_android_device_asus_grouper/commit/fc7cab79fc1c668a4f89e33c5afcb60bf8f99450) - Wi-Fi scanning

This is a commit for an older wifi card, probably some test builds with this CL might give some more insight.

**Comment by ripee** [comment]
> [Exynos7580/android_device_samsung_a3xeltexx@37cb1b1](https://github.com/Exynos7580/android_device_samsung_a3xeltexx/commit/37cb1b1d4213240f7d10c2ba5edd34b68660090f) - hwbinder
> [universal5433/android_device_samsung_trelte-common@2575143](https://github.com/universal5433/android_device_samsung_trelte-common/commit/25751430efc1f9c75f5d0b4493e25941c17bdf80) - EGL crash
> [aperomsik/lineageos_android_device_asus_grouper@fc7cab7](https://github.com/aperomsik/lineageos_android_device_asus_grouper/commit/fc7cab79fc1c668a4f89e33c5afcb60bf8f99450) - Wi-Fi scanning
> 
> This is why I've created those commits.

How about we focus on fixing the hotspot first.

**Comment by JoshuaDoes** [comment]
Pretty much agreeing with the above here. I don't see a need to disable background Wi-Fi scanning (it can be toggled off in the settings if it's not preferential, but it does work on our devices), and honestly rather than outright disabling ScreenLights we should just fix the underlying issue since the intended behavior is still occurring prior to the exception. Also, since AOSP has proved that the two "no controller energy info supplied" messages mean nothing important, we can safely let it be and not try to change what is already known to work. If you can show me a way that these changes (or future changes that you may make) actually fix an underlying issue or enhance the device in some way, I'll be much more open to including them, but this to me just seems like it's abandoning the solution and disabling what already works.
